File: sandbox/fortranpython/navaro/callpy.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from __future__ import print_function
import numpy as np 
import navaro as na

def fonction(i): 
    print('f(i=%d)=%d' %(i, i*i))
    return i*i

s = na.sommef(fonction, 3)  # appelle "fonction(i)"

print("somme =", s)

File: sandbox/fortranpython/navaro/callpy.f90
! from http://calcul.math.cnrs.fr/Documents/Ecoles/2010/f2py-cours.pdf
! Pierre Navaro

! cette fonction appelle une fonction python

subroutine sommef(f, n, s)
    ! Calcule la somme (f(i), i=1,n)
    external f
    integer, intent(in) :: n
    real, intent(out) :: s
    s = 0.0
    do i = 1, n
        s = s + f(i)
    end do
end subroutine sommef
